mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -251,7 +252,7 @@ private boolean textContains(String value, MailboxMessage message) throws IOExce
     }
 
     private InputStream textHeaders(MailboxMessage message) throws MimeIOException, IOException {
-        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        ByteArrayOutputStream out = new ByteArrayOutputStream(Ints.checkedCast(message.getFullContentOctets()));
```

Comment:
What kind of tests would you expect?

This changeset is tested through AstractSearchIndex tests, and they did not broke.